### ISSUE TITLE
Possibly using incorrect bitrate when scaling down to 480p.

### ISSUE THREAD

**Issue by mschout** [donmelton/video_transcoding#58]
Hi.

When using transcode-video with a 1080p source for example, I like to create both an HD and SD version.  For the SD version, I target a max width and height that matches 480p by using args --max-width 854 --max-height 480.  This is the standard width and height of 480p.

The problem is that transcode-video has a block that sets vbv_maxrate this way:

...
      elsif width > 720 or height > 576
        handbrake_options['encoder-level'] = '3.1'
        vbv_maxrate = @vbv_maxrate_720p
      else
        handbrake_options['encoder-level'] = '3.0'
        vbv_maxrate = @vbv_maxrate_480p
      end

Because 480p has a standard width of 854, the "width > 720" matches, and @vbv_maxrate_720p is used instead of @vbv_maxrate_480p

Perhaps it would be better to use

elsif width > 854 or height > 576 

here instead?

**Comment by donmelton** [comment]
@mschout I was wondering when someone would ask for this. :) The reason being that I put a hack in my old Bash-based script for this. But that hack didn't change the width constant. It simply made the test about total number of pixels instead, figuring `720 * 576 = 414720`. Which means `854 * 480 = 409920` would still fit below that threshold.

I'll consider implementing the same hack here. In the meantime, just use the new `--target` option to configure the bitrate, i.e. `--target 2000`.

**Comment by mschout** [comment]
Ok, thanks!

**Comment by donmelton** [comment]
@mschout Your use case is one of the reasons I added the `--target` option -- to make it easier for people using custom widths and heights. :)

**Comment by donmelton** [comment]
@mschout BTW, you don't have to use the `--target` option if you're willing to tell HandBrake to _set_ the width and height of your output directly rather than just _limit_ those values with the `--max-width` and `--max-height` options. Instead of using `854`, you could use `720` or even `640` for the width. That will cause the bitrate targeting algorithm to select `2000` Kbps. For example:

```
transcode-video --handbrake-option width=640 --handbrake-option height=480 "/path/to/Movie.mkv"
```

This, of course, squishes the pixels horizontally but when it's played back, the display aspect ratio which is set automatically for you, will expand the video to `853` by `480`.

And using the entire `720` by `576` limit will be slightly higher resolution, still getting the `2000` Kbs bitrate target:

```
transcode-video --handbrake-option width=720 --handbrake-option height=576 "/path/to/Movie.mkv"
```

Which is, again, squished horizontally but expanded on playback to `1024` by `576` because the display aspect ratio is set automatically for you.

And this trick works even if you crop the video. :)

Let me know whether that's a better solution for you.

**Comment by mschout** [comment]
I guess I don't know, or don't have an opinion on which one is "better".  I'll probably just stick with --max-width 854 --max-height 480 --target 2000 because it is less typing.

**Comment by donmelton** [comment]
@mschout Fair enough! Just wanted to make sure you knew about that. :)

**Comment by donmelton** [comment]
@mschout OK, I have a patch ready for this that I'm testing now. Basically, I just re-implemented the hack from my old Bash-based script. So far, it seems to work. I'll probably release it in a day or two once I get some other fixes complete.

BTW, when you create your SD versions of your collection, you might want to try reducing the size of the audio as well. You could do that with `--ac3-bitrate 384 --pass-ac3-bitrate 448` or you could eliminate surround audio altogether with `--audio-width stereo`. Just a thought.

**Comment by donmelton** [comment]
@mschout OK, the fix has just been released in version 0.8.0. Just update to get it.

Thanks again!
